python/triton/compiler/compiler.py

```diff
@@ -223,7 +228,7 @@ def compile(src, target=None, options=None):
     if ir_source:
         assert isinstance(src, str), "source must be either AST or a filepath"
         context = ir.context()
-        src = IRSource(src, context)
+        src = IRSource(src, context, target)
```

Comment:
Why don't you pass the backend here directly instead of calling make_backend in the contructor?